pure-ftpd: A free, secure, production-quality and standard-conformant FTP server

diff --git a/pkgs/servers/ftp/pure-ftpd/default.nix b/pkgs/servers/ftp/pure-ftpd/default.nix
index e566696..9bcab03 100644
--- a/pkgs/servers/ftp/pure-ftpd/default.nix
+++ b/pkgs/servers/ftp/pure-ftpd/default.nix
@@ -1,13 +1,30 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl, openssl
+, mysqlSupport ? false, mysql ? null
+}:
 
-stdenv.mkDerivation rec {
-  name = "pure-ftpd-1.0.42";
+assert mysqlSupport -> mysql != null;
+
+let
+  pkg = "pure-ftpd";
+  ver = "1.0.42";
+in stdenv.mkDerivation rec {
+  name = "${pkg}-${ver}";
 
   src = fetchurl {
-    url = "https://download.pureftpd.org/pub/pure-ftpd/releases/${name}.tar.gz";
+    url = "https://download.pureftpd.org/pub/pure-ftpd/releases/${pkg}-${ver}.tar.gz";
     sha256 = "1yg7v1l3ng7c08nhh804k28y1f8ccmg0rq1a9l2sg45ib273mrvv";
   };
 
+  buildInputs = [ openssl ]
+    ++ stdenv.lib.optional mysqlSupport mysql;
+
+  configureFlags = ''
+    --with-ftpwho --with-quotas
+    --with-ratios --with-puredb --with-altlog --with-throttling
+    --with-privsep --with-tls
+    ${if mysqlSupport then "--with-mysql=${mysql}" else ""}
+  '';
+
   meta = with stdenv.lib; {
     description = "A free, secure, production-quality and standard-conformant FTP server";
     homepage = https://www.pureftpd.org;
@@ -17,3 +34,5 @@ stdenv.mkDerivation rec {
   };
 
 }
+
+# vim: et ts=2 sw=2 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 23be56b..cf35dd9 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -10647,7 +10647,10 @@ in
 
   pumpio = callPackage ../servers/web-apps/pump.io { };
 
-  pure-ftpd = callPackage ../servers/ftp/pure-ftpd { };
+  pure-ftpd = callPackage ../servers/ftp/pure-ftpd {
+    mysqlSupport = true;
+    mysql = mysql55;
+  };
 
   pyIRCt = callPackage ../servers/xmpp/pyIRCt {};
 
